commit 4bdb90409ff6de07ff161396b3aea5ef30d0e819
Author: aperez@igalia.com <aperez@igalia.com@268f45cc-cd09-0410-ab3c-d52691b4dbfc>
Date:   Fri Aug 11 13:25:57 2017 +0000

    [WPE] Build failure with Clang 4.0.1: no matching conversion for functional-style cast from 'pointer' (aka 'unsigned short *') to 'WTF::String'
    https://bugs.webkit.org/show_bug.cgi?id=175477
    
    Reviewed by Žan Doberšek.
    
    * Shared/wpe/WebEventFactory.cpp:
    (WebKit::singleCharacterStringForKeyEvent): Cast to UChar* to make
    compiler choose the correct WTF::String constructor.
    
    
    git-svn-id: http://svn.webkit.org/repository/webkit/trunk@220585 268f45cc-cd09-0410-ab3c-d52691b4dbfc

diff --git a/Source/WebKit2/Shared/wpe/WebEventFactory.cpp b/Source/WebKit2/Shared/wpe/WebEventFactory.cpp
index 5571e1a..e0ac6ef 100644
--- a/Source/WebKit2/Shared/wpe/WebEventFactory.cpp
+++ b/Source/WebKit2/Shared/wpe/WebEventFactory.cpp
@@ -59,7 +59,7 @@ static String singleCharacterStringForKeyEvent(struct wpe_input_keyboard_event*
     glong length;
     GUniquePtr<gunichar2> uchar16(g_ucs4_to_utf16(&event->unicode, 1, 0, &length, nullptr));
     if (uchar16)
-        return String(uchar16.get());
+        return String(reinterpret_cast<UChar*>(uchar16.get()), length);
     return String();
 }
 
